src/xcode/ENA/ENA/Source/Services/HealthCertificateValidation/BoosterNotificationsService.swift

```diff
@@ -16,19 +21,45 @@ This service is responsible for handling the logic for the Booster Notifications
 Currently it is using the rules download service to fetch the latest Booster Notifications rules
 */
 
-class BoosterNotificationsService {
+class BoosterNotificationsService: BoosterNotificationsServiceProviding {
 	
 	// MARK: - Init
 	
-	init(rulesDownloadService: RulesDownloadServiceProviding) {
+	init(
+		rulesDownloadService: RulesDownloadServiceProviding,
+		validationRulesAccess: BoosterRulesAccessing = ValidationRulesAccess()
+	) {
 		self.rulesDownloadService = rulesDownloadService
+		self.validationRulesAccess = validationRulesAccess
 	}
 	
+	func applyRulesForCertificates(
+		certificates: [DigitalCovidCertificateWithHeader],
+		completion: @escaping (Result<ValidationResult, BoosterNotificationRuleValidationError>) -> Void
+	) {
+		downloadBoosterNotificationRules { [weak self] downloadedRules in
+			guard let self = self else { return }
+			
+			switch downloadedRules {
+			case .success(let rules):
+			
+				let certLogicEngine = CertLogicEngine(rules: rules)
```

Comment:
This is not needed, just omit `certLogicEngine` and it will be initialized internally in the `ValidationRulesAccess` implementation. You can use the `certLogicEngine` API if you want to inject doubles (mocks) in case of unit testing.